modules/cloudsql-instance/variables.tf

```diff
@@ -38,20 +38,22 @@ variable "availability_type" {
 variable "backup_configuration" {
   description = "Backup settings for primary instance. Will be automatically enabled if using MySQL with one or more replicas."
   type = object({
-    enabled            = bool
-    binary_log_enabled = bool
-    start_time         = string
-    location           = string
-    log_retention_days = number
-    retention_count    = number
+    enabled                        = optional(bool)
+    binary_log_enabled             = optional(bool)
+    start_time                     = optional(string)
+    location                       = optional(string)
+    log_retention_days             = optional(number)
+    point_in_time_recovery_enabled = optional(bool)
+    retention_count                = optional(number)
   })
   default = {
-    enabled            = false
-    binary_log_enabled = false
-    start_time         = "23:00"
-    location           = null
-    log_retention_days = 7
-    retention_count    = 7
+    enabled                        = false
+    binary_log_enabled             = false
+    start_time                     = "23:00"
+    location                       = null
+    log_retention_days             = 7
+    point_in_time_recovery_enabled = null
+    retention_count                = 7
```

Comment:
Can you put the defaults as part of the fields definition? This way users can specify only the ones they care about:

```hcl
  type = object({
    enabled                        = optional(bool, false)
    binary_log_enabled             = optional(bool, false)
    start_time                     = optional(string, "23:00")
    location                       = optional(string)
    log_retention_days             = optional(number, 7)
    point_in_time_recovery_enabled = optional(bool)
    retention_count                = optional(number, 7)
  })
  default = {}
```

It might also be a good time to set the variable as non-nullable.

Reply:
Sure, done, thanks for the catch